client/gutenberg/extensions/map-block/info-window/index.js

```diff
@@ -0,0 +1,86 @@
+/**
+ * Wordpress dependencies
+ */
+
+import {
+	Component,
+	createPortal
+} from '@wordpress/element';
+
+/**
+ * External dependencies
+ */
+
+/**
+ * Internal dependencies
+ */
+
+export class InfoWindow extends Component {
+
+	constructor() {
+
+		super( ...arguments );
+		this.closeClick = this.closeClick.bind( this );
+
+	}
+
+	componentDidMount() {
+
+		const { google } = this.props;
+		this.el = document.createElement( 'DIV' );
```

Comment:
Could we avoid any direct DOM manipulations inside components?

Reply:
This is a tricky one. When a map marker is clicked, I open a [Google Maps Info Window](https://developers.google.com/maps/documentation/javascript/infowindows). When you do this, you can pass in plain text, markup, or a DOM node. Because I'm allowing editing of the fields within the Info Window, I need to be able to render React components into it, which I do by creating this DIV and then treating is as a React Portal. So even though that `createElement` looks suspicious, its actually setting up a React pattern - a portal.